pkgs/development/libraries/libspnav/configure-socket-path.patch

```diff
@@ -0,0 +1,39 @@
+diff --git a/spnav.c b/spnav.c
+index f9e10f8..f9a117b 100644
+--- a/spnav.c
++++ b/spnav.c
+@@ -36,7 +36,8 @@ OF SUCH DAMAGE.
+ #include <sys/select.h>
+ #include "spnav.h"
+ 
+-#define SPNAV_SOCK_PATH "/var/run/spnav.sock"
++#define DEFAULT_SPNAV_SOCK_PATH "/var/run/spnav.sock"
++#define SPNAV_SOCK_PATH_ENV "SPNAVD_SOCK_LOCATION"
+ 
+ #ifdef USE_X11
+ #include <X11/Xlib.h>
+@@ -70,6 +71,15 @@ static struct event_node *ev_queue, *ev_queue_tail;
+ /* AF_UNIX socket used for alternative communication with daemon */
+ static int sock = -1;
+ 
++static char *socket_path()
++{
++	char *location;
++	if((location = getenv(SPNAV_SOCK_PATH_ENV))) {
```

Comment:
See comment above - look in `$XDG_RUNTIME_DIR/spnav.sock` if exists, then fall back to the system-wide location.